---
title: "create MAE"
output: html_document
---

```{r}
if (!requireNamespace('BiocManager'))
  install.packages('BiocManager')

if (!requireNamespace('MultiAssayExperiment'))
  BiocManager::install('MultiAssayExperiment')

if (!requireNamespace('data.table'))
  install.packages('data.table')
```
```{r}
library(MultiAssayExperiment)
library(data.table)
library(magrittr)
library(dplyr)
library(stringr)
library(readr)
```

Helper functions:

```{r}
## functions to get number of common samples/features for two datasets
scom <- function(x,y) sum(colnames(x) %in% colnames(y))
fcom <- function(x,y) sum(rownames(x) %in% rownames(y))

## functions to get names of mismatching samples for two datasets
smismatch <- function(x,y) colnames(x)[!colnames(x) %in% colnames(y)]

## pairwise common samples for a list of datasets
scom_list <- function(data_list1, data_list2 = data_list1) sapply(data_list1, function(x) sapply(data_list2, function(y) scom(x, y)))

## pairwise common feartures for a list of datasets
fcom_list <- function(data_list1, data_list2 = data_list1) sapply(data_list1, function(x) sapply(data_list2, function(y)fcom(x, y)))

name_mismatch <- function(data_list1, data_list2=data_list1) lapply(data_list1, function(x) lapply(data_list2, function(y)smismatch(x, y)))

## function to read tsv files into data.frames with first column as rownames
fread_tsv <- function(file_list) {
  out <- lapply(file_list, function(x) as.matrix(data.frame(fread(x), row.names = 1)))
  ## name the list with file name
  names(out) <- gsub(".tsv", "", basename(file_list))
  out
}
```

# To DO

----
* Why are there no metadata for metabolomics data?
* See why metadata is missing a lot of sample names in WGS data, and why there are duplicates

----

# Data

File names:

```{r}
## input
metab_files <- list.files(path = '../data/merged_Metabolomics/', full.names = TRUE)
wgs_files <- list.files(path = '../data/Merged_WGS/', full.names = TRUE)
metadata_file <- '../data/metadata.tsv'
prot_files <- list.files('../data/Proteomics/', full.names = TRUE)
manifest_file <- "../data/hmp_cart_all_ibdmdb.tsv"
```

Read files:

## metadata

```{r}
metadata <- data.frame(fread(metadata_file))
metadata$new_ID <- paste0(metadata$sample_id, "_", metadata$visit_number)
manifest <- data.frame(fread(manifest_file))
```

```{r}
##TODO Why are there duplicate metadata
## duplicate metadata
sum(duplicated(metadata))
```

* from Antoine: the same sample can be observed in different omics files.
In the meantime, the same omics is described in different files (i.e. pathabundance, pathcoverage, ... for WMS).

```{r}
manifest %>% filter(sample_id == "d39c1941c8f6e8b0f6ead5d7f705b0de") %>%
  filter(urls %>% str_detect("wgs/analysis/hmmrc/"))

metadata %>% filter(sample_id == "d39c1941c8f6e8b0f6ead5d7f705b0de") %>% filter(data_type == "WGS_functional")
```

* metaolomics data do have metadata

```{r}
metadata %>% filter(data_type == "Metabolomics") %>% head
```

## Metabolomic

former metabo
```{r, eval=F}
former_metab_files<- c("~/ls3x/Nextcloud/files/IBDMDB/ibd/metabolome/microbiome/analysis/new_metabo/all_C18_neg_compound.tsv","~/ls3x/Nextcloud/files/IBDMDB/ibd/metabolome/microbiome/analysis/new_metabo/all_C8_pos_compound.tsv", "~/ls3x/Nextcloud/files/IBDMDB/ibd/metabolome/microbiome/analysis/new_metabo/all_HILIC_neg_compound.tsv", "~/ls3x/Nextcloud/files/IBDMDB/ibd/metabolome/microbiome/analysis/new_metabo/all_HILIC_pos_compound.tsv")
former_metabo <- lapply(former_metab_files, readr::read_tsv)
names(former_metabo) <- str_remove(former_metab_files, ".*/all_") %>% str_remove("_compound.tsv")
save(former_metabo, file = "../data/metabo_former.RData")
```

```{r}
download.file(url = "http://adcloud.genome.ulaval.ca/index.php/s/Xd6CY9pLNc2SRQC/download", destfile = "../data/wgs.RData", method = "wget")
load("../data/metabo_former.RData")
```

* change colnames

```{r}
change_metabo_colnames <- function(x){
  manifest_metabo <- manifest %>% filter(str_detect(urls, "metabolome")) %>% 
    mutate(ID = str_remove(urls, ".*/") %>% str_remove("_.*")) %>%
    dplyr::select(ID, sample_id) %>% unique
  
  # new_ID is now samppleid_visit_number
  metadata_metabo <- metadata %>% filter(data_type == "Metabolomics") %>%
    dplyr::select(sample_id, new_ID) %>% unique
  
  new_colnames <- colnames(x)[-1] %>% as.data.frame %>%  purrr::set_names("name") %>%
    left_join(manifest_metabo, by = c("name"="ID")) %>% # add sample_id
    left_join(metadata_metabo, by = c("sample_id" = "sample_id")) # add new_ID
  
  colnames(x) <- c("Compound", new_colnames$new_ID)
  x <- x %>% column_to_rownames("Compound")
  return(x)
}

former_metabo <- lapply(former_metabo,change_metabo_colnames)
```




```{r}
## read files as named list
metab_list <- fread_tsv(metab_files)
```


```{r}
## unintended X added to colnames with subject_id that start with numbers
lapply(metab_list, function(x) grep(pattern = "^X", colnames(x)))
## drop it
metab_list <- lapply(metab_list, function(x) set_colnames(x, gsub(pattern = "^X",replacement = "", colnames(x))))
```

Note : Why don't use `as.data.frame` instead of `data.frame` in the `fread_tsv` function or `data.frame(check.names = FALSE, ...)` to get rid of `X...`?

Check all samples have metadata:

```{r}
##TODO There are no metadata available for these samples
## number of samples in each dataset not present in metadata$new_ID
lapply(metab_list, function(x) sum(! colnames(x) %in% metadata$new_ID))
lapply(former_metabo, function(x) sum(! colnames(x) %in% metadata$new_ID))

## number of trimmed sample names (no visit) in each dataset not present in metadata$sample_id
lapply(metab_list, function(x) sum(! gsub(pattern = "_.*$", "", x = colnames(x)) %in% metadata$sample_id))
## so all samples are valid, only their visit record is not there
```

```{r}
View(metab_list[[1]][1:10,1:10])
```

```{r}
## check that new ID is same as subjectid_visitnumber
sum(metadata$new_ID != paste0(metadata$subject_id, "_", metadata$visit_number))
```

```{r}
## common samples
scom_list(metab_list)
scom_list(former_metabo) # ok
## common features
fcom_list(metab_list)
fcom_list(former_metabo) # ok
```

```{r}
## adding 'metab_' to names to distinguish in MAE object
names(metab_list) %<>% paste0("metab_", .)
names(former_metabo) %<>% paste0("metab_", .)
```


## WGS

```{r}
#example of mismatch, with no ECS here
manifest %>% filter(str_detect(urls, "CSM7KOMV.*tsv"))
```

```{r, eval =F}
# former colnames; files in Cloud folder
WGS_former_files <- c("~/ls3x/Nextcloud/files/IBDMDB/ibd/genome/microbiome/wgs/analysis/hmmrc/new_files/ALL_pathabundance.tsv", "~/ls3x/Nextcloud/files/IBDMDB/ibd/genome/microbiome/wgs/analysis/hmmrc/new_files/ALL_pathcoverage.tsv",
"~/ls3x/Nextcloud/files/IBDMDB/ibd/genome/microbiome/wgs/analysis/hmmrc/new_files/ALL_genefamilies.tsv",
"~/ls3x/Nextcloud/files/IBDMDB/ibd/genome/microbiome/wgs/analysis/hmmrc/new_files/ALL_ecs.tsv", 
"~/ls3x/Nextcloud/files/IBDMDB/ibd/genome/microbiome/wgs/analysis/hmscp/new_files/ALL_taxonomicprofile.tsv")
WGS_former_data <- lapply(WGS_former_files, readr::read_tsv)
names(WGS_former_data) <- WGS_former_files %>% str_remove(".*_") %>% str_remove(".tsv")

save(WGS_former_data, file = "../data/wgs.RData")
```

```{r}
download.file(url = "http://adcloud.genome.ulaval.ca/index.php/s/6pj7r6nG2qnyLXw/download", destfile = "../data/wgs.RData", method = "wget")
load("../data/wgs.RData")
```

* change column

```{r}
change_wgs_colnames <- function(x){
  manifest_wgs <- manifest %>% filter(str_detect(urls, "/wgs/")) %>% 
    mutate(ID = str_remove(urls, ".*/") %>% str_extract(".*_") %>% str_remove("_$")) %>%
    dplyr::select(ID, sample_id) %>% unique
  
  # new_ID is now samppleid_visit_number
  metadata_wgs <- metadata %>% filter(str_detect(data_type, "WGS")) %>%
    dplyr::select(sample_id, new_ID) %>% unique
  
  new_colnames <- colnames(x)[-1] %>% str_extract("^[:alnum:]*(_[:alnum:]*)?") %>% as.data.frame %>%  purrr::set_names("name") %>%
    left_join(manifest_wgs, by = c("name"="ID")) %>% # add sample_id
    left_join(metadata_wgs, by = c("sample_id" = "sample_id")) # add new_ID
  
  colnames(x) <- c("XX1", new_colnames$new_ID)
  x <- x %>% column_to_rownames("XX1")
  return(x)
}

former_wgs <- lapply(WGS_former_data,change_wgs_colnames)
```


Same with WGS data:

```{r}
## read files as named list
wgs_list <- fread_tsv(wgs_files)
```

```{r}
## unintended X added to colnames with subject_id that start with numbers
lapply(wgs_list, function(x) grep(pattern = "^X", colnames(x)))
## drop it
wgs_list <- lapply(wgs_list, function(x) set_colnames(x, gsub(pattern = "^X",replacement = "", colnames(x))))
```

```{r}
View(wgs_list[[3]][1:10,1:10])
```

```{r}
## common samples
scom_list(wgs_list)
scom_list(former_wgs)
# common features
fcom_list(wgs_list)
fcom_list(former_wgs)
```

What are the non-matching ones?

```{r}
wgs_mismatch <- name_mismatch(wgs_list)
wgs_mismatch_former <- name_mismatch(former_wgs)
```

from Antoine: I think we should drop ecs and the sample `1419f08f554e0c93f3b62fe90c246a08_13`
```{r}
## one-by-one mismatches can be inspected as follows:
## for pathcoverage and pathabundance
wgs_mismatch$pathcoverage$pathabundance
## this suggests there's been 2 NA's in headers of one dataset

wgs_mismatch$pathcoverage$WGS_community
## this suggests there's been 2 NA's in headers of one dataset, plus an extra sample
## which does not exist in metadata:
"1419f08f554e0c93f3b62fe90c246a08_13" %in% metadata$new_ID
```

The main mismatch comes from ecs (~210 samples):

```{r}
## those from genefamilies not present in ecs
head(wgs_mismatch$genefamilies$ecs)
```

 Are these valid new_ID's?
 
```{r}
##TODO There are no metadata available for these samples
## number of samples in each dataset not present in metadata$new_ID
lapply(wgs_list, function(x) sum(! colnames(x) %in% metadata$new_ID))
```

Doesn't look like it


```{r}
## number of genefamilies samples present in metadata
sum(colnames(wgs_list$genefamilies) %in% metadata$new_ID)
```

Shared samples with metabolites:

```{r}
## common samples b/w WGS (~300) and metabolomics (545)
scom_list(wgs_list, metab_list)
scom_list(former_wgs, former_metabo)
```

[#Antoine] we noticed that there were `NA`s in original files that lead to weird namings by `fread`. Also you can see the `ecs` data does not share a lot of samples with others. Can you please let me know if I have to i) drop `ecs` dataset and ii) drop some non-matching samples from other datasets?

From Antoine, I would probably drop `ecs`

```{r}
lapply(former_wgs, function(x) sum(is.na(colnames(x))))
# NA in genefamilies was "CSM67UEW_TR" wich is missing from menifest file
manifest %>% filter(str_detect(urls, "CSM67UEW_TR"))
```

from Antoine, we should probably remove this missing sample.

I'm assuming these are separate data so I don't worry about common features.

```{r}
## adding 'WGS_' to names to distinguish in MAE object
## exclude WGS_community
names(wgs_list) <- gsub(".tsv", "", basename(wgs_files))
names(wgs_list) %<>% paste0("WGS_", .)
## last one does already have WGS prefix
names(wgs_list)[5] <- gsub(".tsv", "", basename(wgs_files)[5])
names(wgs_list)

names(former_wgs) %<>% paste0("WGS_", .)
```

## Proteomic

```{r}
## read files as named list
prot_list <- fread_tsv(prot_files)
```

```{r}
## this one seems to use new_ID = paste0(sample_id,"_",visit)?
View(prot_list[[1]][1:10,1:10])
```

```{r}
## common samples
scom_list(prot_list)
## common features
fcom_list(prot_list)
```

[#Antoine] So I'm going with the one from `pep` ones. Also keeping the `kos` one as separate dataset, right?

from Antoine: As wgs `ecs`, I would drop `kos`.


```{r, eval = F}
former_prot_files <- list("~/ls3x/Nextcloud/files/IBDMDB/ibd/proteome/microbiome/analysis/new_files/ALL_1pep.1p.tsv","~/ls3x/Nextcloud/files/IBDMDB/ibd/proteome/microbiome/analysis/new_files/ALL_1pep.5p.tsv","~/ls3x/Nextcloud/files/IBDMDB/ibd/proteome/microbiome/analysis/new_files/ALL_2pep.1p.tsv", "~/ls3x/Nextcloud/files/IBDMDB/ibd/proteome/microbiome/analysis/new_files/ALL_2pep.5p.tsv")
former_prot <- lapply(former_prot_files, readr::read_tsv)

names(former_prot) <- former_prot_files %>% str_remove(".*_") %>% str_remove(".tsv")
save(former_prot, file = "../data/prot.RData")
```

```{r}
download.file(url = "http://adcloud.genome.ulaval.ca/index.php/s/GgFwJrRNMtBqRp2/download", destfile = "../data/prot.RData", method = "wget")
load("../data/prot.RData")
```

* change prot colnames

```{r}
change_prot_colnames <- function(x){
  manifest_prot <- manifest %>% filter(str_detect(urls, "proteome")) %>% 
    mutate(ID = str_remove(urls, ".*/") %>% str_remove("_.*")) %>%
    dplyr::select(ID, sample_id) %>% unique
  
  # new_ID is now samppleid_visit_number
  metadata_prot <- metadata %>% filter(str_detect(data_type, "Proteomics")) %>%
    dplyr::select(sample_id, new_ID) %>% unique
  
  new_colnames <- colnames(x)[-1] %>% as.data.frame %>%  purrr::set_names("name") %>%
    left_join(manifest_prot, by = c("name"="ID")) %>% # add sample_id
    left_join(metadata_prot, by = c("sample_id" = "sample_id")) # add new_ID
  
  colnames(x) <- c("XX1", new_colnames$new_ID)
  x <- x %>% column_to_rownames("XX1")
  return(x)
}

former_prot <- lapply(former_prot,change_prot_colnames)
```

```{r}
## keeping 2pep_5p_FDR_5ppm and kos datasets
prot_list <- prot_list[4:5]

## adding 'prot_' to names to distinguish in MAE object
names(prot_list) %<>% paste0("prot_", .)
```

```{r}
##TODO There are no metadata available for these samples
## number of samples in each dataset not present in metadata$new_ID
lapply(prot_list, function(x) sum(! colnames(x) %in% metadata$new_ID))
lapply(former_prot, function(x) sum(! colnames(x) %in% metadata$new_ID))
```

Create MAE object:

```{r}
iHMP <- MultiAssayExperiment(experiments = ExperimentList(c(prot_list, wgs_list, metab_list))
                     #, colData = data.frame(metadata, row.names = "new_ID") ## this should use a row.name that matches the sample names - not possible at the moment because there are a lot of duplicated and we don't know which ones we wanna use
                     # as sample names
                     )
```


